lib/ansible/plugins/lookup/rabbitmq.py

```diff
@@ -0,0 +1,175 @@
+# (c) 2018, John Imison <john+github@imison.net>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = """
+    lookup: rabbitmq
+    author: John Imison <@Im0>
+    version_added: "2.7"
+    short_description: Retrieve messages from an AMQP/AMQPS RabitMQ queue/channel.
+    description:
+        - This lookup uses a basic get to retrieve all, or a limited number C(count), messages from a RabbitMQ queue.
+    options:
+      url:
+        description:
+          - An URI connection string to connect to the AMQP/AMQPS RabbitMQ server.
+          - For more information refer to the URI spec U(https://www.rabbitmq.com/uri-spec.html).
+        required: True
+      channel:
+        description:
+          - The channel/queue to get messages from.
+        required: True
+      count:
+        description:
+          - How many messages to collect from the queue.
+          - If not set, defaults to retrieving all the mesages from the queue.
+        default: All messages
+    requirements:
+        - The python pika package U(https://pypi.org/project/pika/).
+    notes:
+        - This lookup implements BlockingChannel.basic_get to get message from a RabbitMQ server.
+        - Pika is a pure-Python implementation of the AMQP 0-9-1 protocol that tries to stay fairly independent of the underlying network support library.
+        - More information about pika can be found at U(https://pika.readthedocs.io/en/stable/).
+        - This plugin is tested against RabbitMQ.  Other AMQP 0.9.1 protocol based servers may work but not tested/gaurenteed.
+"""
+
+
+EXAMPLES = """
+- name: Get all messages off a queue
+  debug:
+    msg: "{{ lookup('rabbitmq', url='amqp://guest:guest@192.168.0.10:5672/%2F', channel='hello') }}"
+
+
+# If you are intending on using the returned messages as a variable in more than one task, it is recommended to set_fact.
```

Comment:
Should not that (messages from channel/queue are consumed) be stated in `notes` metadata ? This comment should mention that messages are consumed each time the lookup is called.

Reply:
Thanks.  I hope this time the update is appropriate.